crates/kitsune_p2p/kitsune_p2p/src/spawn/actor.rs

```diff
@@ -27,27 +27,34 @@ mod space;
 use ghost_actor::dependencies::tracing;
 use space::*;
 
+type EvtRcv = futures::channel::mpsc::Receiver<KitsuneP2pEvent>;
+type KSpace = Arc<KitsuneSpace>;
+type KAgent = Arc<KitsuneAgent>;
+type KBasis = Arc<KitsuneBasis>;
```

Comment:
I'd rather re-arange the `Kitsune*` types so the `Arc`s are internal so you don't have to specify them everywhere... but that feels out of scope to this PR.

Reply:
I'm a little concerned about that plan, we already have at least one occurrence of `Arc<Vec<Arc<KitsuneOpHash>>>` and that vec could have thousands of items in it. I don't know how much overhead `Arc` incurs, I know it's small but it's not zero-cost. I believe in that case it should just be `Arc<Vec<KitsuneOpHash>>` which would be impossible with that rearrangement.

Reply:
I disagree... remember, every one of these keys is 32 bytes long. Arc adds 16 bytes (two AtomicUsizes on a u64 system) ONCE. Without the arc, you use up 32 bytes of memory on the system for every single clone.

Just because we put it in an `Arc<Vec<_>>`, it doesn't mean we want to start cloning those underlying bytes... we still likely have other copies of that data elsewhere in the system.